onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -997,18 +997,68 @@ def aten_conv1d(
     raise NotImplementedError()
 
 
+@torch_op("aten::conv2d", trace_only=True)
 def aten_conv2d(
-    input: TensorType,
-    weight: TensorType,
-    bias: Optional[TensorType] = None,
+    input: TFloat,
+    weight: TFloat,
+    bias: Optional[TFloat] = None,
     stride: Sequence[int] = (1, 1),
     padding: Sequence[int] = (0, 0),
     dilation: Sequence[int] = (1, 1),
     groups: int = 1,
-) -> TensorType:
+) -> TFloat:
     # conv2d(Tensor input, Tensor weight, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] dilation=1, int groups=1) -> Tensor
 
-    raise NotImplementedError()
+    # Attributes need to be manipulated in Python to match ONNX's conv2d
+    if not isinstance(padding, Sequence):
+        padding = (padding, padding)
+    pads = [*padding, *padding]
+
+    if not isinstance(dilation, Sequence):
+        dilation = (dilation, dilation)
+    dilations = list(dilation)
+
+    if not isinstance(stride, Sequence):
+        stride = (stride, stride)
+    strides = list(stride)
+
+    if bias is None:
+        weight_dim_0 = op.Gather(op.Shape(weight), 0, axis=0)
+        bias_shape = op.Expand(weight_dim_0, op.Constant(value_ints=[1]))
+        zero = op.CastLike(0.0, input)
+        bias = op.Expand(zero, bias_shape)
+
+    result = aten_conv2d_onnx(
+        input, weight, bias, strides=strides, pads=pads, dilations=dilations, groups=groups
+    )
+
+    return result
+
+
+@torch_op("aten::conv2d", overload=True)
+def aten_conv2d_onnx(
```

Comment:
So if I am building graph from scratch, I should use `aten_conv2d_onnx` instead?

Reply:
Should we rename aten_conv2d_onnx to _aten_conv2d_onnx to avoid the confusion like Titai had?

Reply:
SG

Reply:
Done